sdk/identity/Azure.Identity/src/IdentityClient.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Pipeline.Policies;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    internal class IdentityClient
+    {
+        private readonly IdentityClientOptions _options;
+        private readonly HttpPipeline _pipeline;
+        
+
+        public IdentityClient(IdentityClientOptions options = null)
+        {
+            _options = options ?? new IdentityClientOptions();
+
+            _pipeline = HttpPipeline.Build(_options,
+                    _options.ResponseClassifier,
+                    _options.RetryPolicy,
+                    ClientRequestIdPolicy.Singleton,
+                    _options.LoggingPolicy,
+                    BufferResponsePolicy.Singleton);
+        }
+
+        public async Task<AuthenticationResponse> AuthenticateAsync(string tenantId, string clientId, string clientSecret, IEnumerable<string> scopes, CancellationToken cancellationToken = default)
+        {
+            Uri authUri = new Uri(_options.AuthorityHost + tenantId + "/oauth2/token");
+
+            using (var request = _pipeline.CreateRequest())
+            {
+                request.SetRequestLine(HttpPipelineMethod.Put, authUri);
+
+                ReadOnlyMemory<byte> content = Serialize(
```

Comment:
Sorry, I should've been clearer here. I was talking about having a method that would take `clientId`, `clientSecret` and `scopes` and write directly into JsonWriter avoiding the additional overhead.

Reply:
This change still allocates an array, but no longer allocates a dictionary.  There will be other overloads of Authenticate in the near future which have disparate properties set in their body so I wanted to leave Serialize as generic as possible.   Do you think that the array allocation is something that we need to optimize out?

Reply:
Old habits die hard. We can do it later if we see allocations in Identity client being a problem.